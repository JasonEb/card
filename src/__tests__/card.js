import React from "react"
import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"

import Card from "../components/card/card"

describe("Card", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Card />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  beforeEach( () => {
    return render(<Card />)  
  })

  it("it should have my Name", () => {
      const name = screen.getByText('jason ebueng')
      expect(name).toBeTruthy()
  })

  test("it should have my Linkedin", () => {
    const url = screen.getByText('linkedin.com/in/jasonebueng/')
    expect(url).toBeTruthy()
  })

  test("it should have my GitHub", () => {
    const github = screen.getByText('github.com/JasonEb')
    expect(github).toBeTruthy()
  })

  test("it should have my email", () => {
    const email = screen.getByText('jason.ebueng[at]gmail.com')
    expect(email).toBeTruthy()
  })

  test("it should have my blog", () => {
    const blog = screen.getByText('https://jasoneb.github.io/blog')
    expect(blog).toBeTruthy()
  })
})