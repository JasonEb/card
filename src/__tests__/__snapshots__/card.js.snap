// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandScape Card renders correctly 1`] = `
<div
  className="sc-bdfBwQ fJfIjm"
  style={
    Object {
      "fontFamily": undefined,
    }
  }
>
  <div
    className="sc-gsTCUz czNAIZ"
  >
    <div
      className="sc-dlfnbm frclaY"
    >
      <div
        className="sc-hKgILt fLFlvy"
      >
        <div
          className="sc-eCssSg hugDgv"
        >
          jason ebueng
        </div>
        <div
          className="sc-jSgupP fhWkTJ"
        >
          Web Developer
        </div>
      </div>
    </div>
    <div
      className="sc-gKsewC ioLATl"
    >
      <div
        className="sc-iBPRYJ hCAMas"
      >
        <ul
          className="sc-fubCfw hwrzrL"
        >
          <li
            className="sc-pFZIQ dFaSce"
          >
            <div
              className="sc-jrAGrp eqbWjJ"
            >
              🏠 
            </div>
            Southern California
          </li>
          <li
            className="sc-pFZIQ dFaSce"
          >
            <div
              className="sc-jrAGrp eqbWjJ"
            >
              📧 
            </div>
            jason.ebueng[at]gmail.com
          </li>
          <li
            className="sc-pFZIQ dFaSce"
          >
            <div
              className="sc-jrAGrp eqbWjJ"
            >
              🐈 
            </div>
            github.com/JasonEb
          </li>
          <li
            className="sc-pFZIQ dFaSce"
          >
            <div
              className="sc-jrAGrp eqbWjJ"
            >
              📄 
            </div>
            linkedin.com/in/jasonebueng/ 
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
